### ISSUE TITLE
After updating to 2021.8.0, TV no longer pairs with HomeKit

### ISSUE THREAD

**Issue by bcmitri** [ollo69/ha-samsungtv-smart#93]
**Describe the bug**
After updating to Home Assistant version 2021.8.0, the media player entity created for my Samsung Smart TV by the Samsung TV Smart component no longer seems to be compatible with Home Assistant's native HomeKit integration.  Within the Home app, the device was listed as Not Responding and attempts to remove it and re-pair it result in an error.

**Expected behavior**
I expected the device to continue to function correctly, as it always had prior to upgrading to the latest version of Home Assistant.  When removing and re-pairing the device, I expected the process to complete successfully, without error.

**Screenshots**
![image](https://user-images.githubusercontent.com/53234043/128507548-eeeeef69-5904-43f6-9225-880c3074e4cd.png)

![image](https://user-images.githubusercontent.com/53234043/128507533-5b9335fa-1337-49e3-8c12-bbb83629b1bb.png)

![image](https://user-images.githubusercontent.com/53234043/128507564-08818395-1f52-43c4-8871-e316e9f05ae2.png)

**Environment details:**
 - Home Assistant version installed: 2021.8.2 (but issue was introduced in 2021.8.0)
 - Component version installed: v0.3.36 (latest)
 -  Last know working Home Assistant version: 2021.7.4
 - Last know working component version: v0.3.36 (latest)
 - TV model: UN65TU8500FXZC
 - Environment (HASSIO, Raspbian, etc): 

version | core-2021.8.2
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.9.6
os_name | Linux
os_version | 5.10.53
arch | x86_64
timezone | America/Halifax

**Output of HA logs**
```
Logger: homeassistant.config_entries
Source: components/homekit/accessories.py:250
First occurred: 8:29:43 (1 occurrences)
Last logged: 8:29:43

Error setting up entry Basement TV:21065 for homekit
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 293, in async_setup
    result = await component.async_setup_entry(hass, self)  # type: ignore
  File "/usr/src/homeassistant/homeassistant/components/homekit/__init__.py", line 302, in async_setup_entry
    await homekit.async_start()
  File "/usr/src/homeassistant/homeassistant/components/homekit/__init__.py", line 685, in async_start
    if not await self._async_create_accessories():
  File "/usr/src/homeassistant/homeassistant/components/homekit/__init__.py", line 796, in _async_create_accessories
    acc = self._async_create_single_accessory(entity_states)
  File "/usr/src/homeassistant/homeassistant/components/homekit/__init__.py", line 775, in _async_create_single_accessory
    acc = get_accessory(self.hass, self.driver, state, STANDALONE_AID, conf)
  File "/usr/src/homeassistant/homeassistant/components/homekit/accessories.py", line 206, in get_accessory
    return TYPES[a_type](hass, driver, name, state.entity_id, aid, config)
  File "/usr/src/homeassistant/homeassistant/components/homekit/type_media_players.py", line 223, in __init__
    super().__init__(
  File "/usr/src/homeassistant/homeassistant/components/homekit/type_remotes.py", line 86, in __init__
    super().__init__(*args, category=CATEGORY_TELEVISION, **kwargs)
  File "/usr/src/homeassistant/homeassistant/components/homekit/accessories.py", line 250, in __init__
    firmware_revision=sw_version[:MAX_VERSION_LENGTH],
TypeError: 'NoneType' object is not subscriptable
```

**Additional context**
Because this error arose from upgrading Home Assistant and not the Samsung TV Smart component, I cannot be sure where the issue lies.  I do know there were some HomeKit changes in Home Assistant version 2021.8.0, but I also know that other TVs I have in Home Assistant are not affected by this issue.

**Comment by ollo69** [comment]
I'm not aware of any changes in `media_player` entity, so the issue have to be fixed in `homekit`.
I don't know how it retrieve the array `sw_version`, I think it should check that variable is not `None`. 
Anyway if you discover that some new specific property should be mandatory created by the `media_player` entity, come back here and report.

**Comment by ollo69** [comment]
I found a similar [issue](https://github.com/home-assistant/core/issues/54024). Should be fixed by 2021.8.2, but seems that is not.

**Comment by bcmitri** [comment]
Looks like it was fixed in 2021.8.3.
